### PR TITLE
feat: add new jss utilities and improve performance of old ones

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description
After noticing signficiant performance hits from using `ensureDesignSystemDefaults` and `withDesignSystemDefaults`, this change makes available simple accessors of design system properties. Performance improvements are significant between using these accessors and using `ensureDesignSystemDefaults` to ensure valid values exist.

## Motivation & context
Performance

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-jss-utilities/README.md [hunk 1]
```diff
@@ -193,4 +193,29 @@ function getMyColor(designSystem) {
...
+The `toString` function is used to convert the return value of a function that accepts the design system to a string.
+
+```ts
+{
+    opacity: toString((designSystem) => designSystem.disabledOpacity)
+}
+```
+
+### add / subtract / multiply / divide
+the `add`, `subtract`, `multiply`, and `divide` functions is used to operate on numbers or functions that accept design-systems and return numbers. They accept any number of arguments and perform their operations from left to right, starting with the first argument. eg, subtract(10, 2) => 10 - 2 => 8
```
Comment (chrisdholt) [inline packages/fast-jss-utilities/README.md]: `the `add`, `subtract`, `multiply`, and `divide` functions *are*`

File: packages/fast-jss-utilities/src/math.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+// export type MathUtilityArgumentList<T> = Array<number | ((designSystem: T) => number)>;
```
Comment (janechu) [inline packages/fast-jss-utilities/src/math.ts]: still needed?

File: packages/fast-jss-utilities/src/math.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+export type MathUtilityFunction<T> = (
+    ...args: Array<number | ((designSystem: T) => number)>
+) => (designSystem?: T) => number;
+
+function performOperation<T>(
+    opperation: (a: number, b: number) => number
```
Comment (janechu) [inline packages/fast-jss-utilities/src/math.ts]: operation

File: packages/fast-jss-utilities/src/math.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+) => (designSystem?: T) => number {
+    return (
+        ...args: Array<number | ((designSystem: T) => number)>
+    ): ((designSystem?: T) => number) => {
+        return (designSystem?: T): number => {
+            const firstArg: number | ((designSystem: T) => number) = args[0];
+            let value: number =
+                typeof firstArg === "function" ? firstArg(designSystem) : firstArg;
+
+            for (let i: number = 1; i < args.length; i++) {
```
Comment (janechu) [inline packages/fast-jss-utilities/src/math.ts]: Suggest using for... of

Comment (nicholasrice) [reply packages/fast-jss-utilities/src/math.ts]: can't - I need to start at 1 and not 0

**Pull request merged**